orchagent/saihelper.cpp

```diff
@@ -800,3 +827,270 @@ void handleSaiFailure(bool abort_on_failure)
         abort();
     }
 }
+
+
+inline void initSaiRedisCounterEmptyParameter(sai_s8_list_t &sai_s8_list)
+{
+    sai_s8_list.list = nullptr;
+    sai_s8_list.count = 0;
+}
+
+inline void initSaiRedisCounterEmptyParameter(sai_redis_flex_counter_group_parameter_t &flex_counter_group_param)
+{
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.poll_interval);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.operation);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.stats_mode);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.plugin_name);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.plugins);
+}
+
+inline void initSaiRedisCounterParameterFromString(sai_s8_list_t &sai_s8_list, const std::string &str)
+{
+    if (str.length() > 0)
+    {
+        sai_s8_list.list = (int8_t*)const_cast<char *>(str.c_str());
```

Comment:
i think this is bad practice, sine you need to make sure that passed str will not get out of scope when s8 list will be used

Reply:
I understand it's not a best practice but it doesn't look like there is another way to initialize sai_s8_list from a std::string. we have to use passing by reference here.
maybe we can make all the functions to initialize parameters (starting with `initSaiRedis`) `static` to avoid being called outside the file.
all the functions calling them are like this

1. call `initSaiRedisXXX` to initialize sai_s8_list from std::string
2. call `notifySyncdXXXX` to notify SAI redis

both std::string and sai_s8_list variables are local to the function and are safe.

how do you think or any suggestions? @kcudnik

Reply:
making those static should be better

Reply:
Done.